app-runner/Private/DeviceProviders/SwitchProvider.ps1

```diff
@@ -33,17 +33,44 @@ class SwitchProvider : DeviceProvider {
         $this.Commands = @{
             'connect'            = @($this.TargetControlTool, 'connect --force')
             'disconnect'         = @($this.TargetControlTool, 'disconnect')
-            'poweron'            = @($this.TargetControlTool, 'power-on')
+            'poweron'            = @($this.TargetControlTool, 'power-on -t {0}')
             'poweroff'           = @($this.TargetControlTool, 'power-off')
             'press-power-button' = @($this.TargetControlTool, 'press-power-button')
             'reset'              = @($this.TargetControlTool, 'reset')
             'getstatus'          = @($this.TargetControlTool, 'get-default --detail --json')
+            'detect-target'      = @($this.TargetControlTool, 'detect-target --detail --json')
             'launch'             = @($this.ApplicationRunnerTool, '"{0}" {1}')
             'screenshot'         = @($this.TargetControlTool, 'take-screenshot --directory "{0}" --file-name "{1}"')
             'test-internet'      = @($this.TargetControlTool, 'devmenu -- network confirm-internet-connection')
         }
     }
 
+    [void] HandleMissingDefaultDevkit() {
+        Write-Warning 'No default devkit found. Attempting to detect available devkits on the network...'
+
+        # Run detect-target to find available devkits
+        $detectResult = $this.InvokeCommand('detect-target', @())
+
+        if ([string]::IsNullOrWhiteSpace($detectResult)) {
+            throw 'No default devkit is set and no devkits were detected on the network.'
+        }
+
+        $detectedTargets = $detectResult | ConvertFrom-Json
+        Write-Debug "Detected targets: $($detectedTargets | ConvertTo-Json -Compress)"
+
+        # Extract target name
+        $targetId = $detectedTargets[0].Name
+        if ([string]::IsNullOrWhiteSpace($targetId)) {
+            throw 'No detected Devkit on the network.'
+        }
+
+        Write-Warning "Attempting to power on devkit at $targetId..."
+        $this.InvokeCommand('poweron', @($targetId))
+        Start-Sleep -Seconds 5
+        Write-Warning "Attempting to connect to devkit at $targetId..."
+        $this.InvokeCommand('connect', @($targetId))
```

Comment:
This function should not try to wake up/reboot devkit - that's already handled as part of GetDeviceStatus - which tries to recover the devkit from any state.

Additionally, 'poweron' actually restarts a device that's already on - this is pretty slow and async as you've noticed. That's why the original code tried to get the device status before restarting.